common/nodejs-utils/src/buildplan.rs

```diff
@@ -0,0 +1,133 @@
+use libcnb_data::buildpack_plan::BuildpackPlan;
+
+#[derive(Debug, Default, PartialEq)]
+pub struct NodeBuildScriptsMetadata {
+    pub enabled: Option<bool>,
+}
+
+pub const NODE_BUILD_SCRIPTS_BUILD_PLAN_NAME: &str = "node_build_scripts";
+const NODE_BUILD_SCRIPTS_METADATA_ENABLED_KEY: &str = "enabled";
+
+pub fn read_node_build_scripts_metadata(
+    buildpack_plan: &BuildpackPlan,
+) -> Result<NodeBuildScriptsMetadata, NodeBuildScriptsMetadataError> {
+    buildpack_plan
+        .entries
+        .iter()
+        .filter(|entry| entry.name == NODE_BUILD_SCRIPTS_BUILD_PLAN_NAME)
+        .try_fold(
```

Comment:
That's what it looks like to me Josh, last one wins.

I wonder if always allowing a `enabled = true` to win would be a better behavior. Since:

-  the default is to be enabled
- typically `enabled = false` would be the special override config
- then something explicitly setting `enabled = true` would be to ensure it happens even though something else disabled it.

Reply:
Could also throw an error if we find mixed `true` and `false` values, since it's sort of contradictory / non-deterministic. 

Preferring `enabled = true` is fine with me too.

Reply:
Regarding
> I wonder if always allowing a enabled = true to win would be a better behavior.

and
> Could also throw an error if we find mixed true and false values

I'm not sure I understand how you both are thinking about the build plan here. The entries are ordered so information given by later buildpacks should be given priority. If multiple buildpacks end up using the `node_build_scripts` build plan and set the `enabled` value then the last should win. This is not unlike how we treat environment variables.

The setting is also modelled as an `Option<bool>` to indicate if this value was even set at all in any of the matching build plan entries. This will help distinguish between:
- Some(true) - a participating buildpack is explicitly requesting that scripts be executed
- Some(false) - a participating buildpack is explicitly requesting that scripts should not be executed
- None - the default behavior is used